tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/nvptx_backend_lib.cc

```diff
@@ -329,16 +304,30 @@ StatusOr<string> CompileModuleToPtx(llvm::Module* module,
 
   // Try to fetch the target triple from the module. If not present, set a
   // default target triple.
-  llvm::Triple target_triple = llvm::Triple(module->getTargetTriple());
   if (target_triple.getArch() == llvm::Triple::UnknownArch) {
     LOG(WARNING) << "target triple not found in the module";
-    target_triple = llvm::Triple("nvptx64-unknown-unknown");
+    if (isNVPTX) {
+      target_triple = llvm::Triple("nvptx64-unknown-unknown");
+    } else {
+      target_triple = llvm::Triple("amdgcn--amdhsa-amdgiz");
```

Comment:
Are two dashes in "amdgcn--amdhsa-amdgiz" intended?

Reply:
Yes it's intended.